plugins/opsgenie/icon_opsgenie/actions/get_on_calls/action.py

```diff
@@ -0,0 +1,17 @@
+import insightconnect_plugin_runtime
+from .schema import GetOnCallsInput, GetOnCallsOutput, Input, Component
+
+
+class GetOnCalls(insightconnect_plugin_runtime.Action):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="get_on_calls", description=Component.DESCRIPTION, input=GetOnCallsInput(), output=GetOnCallsOutput()
+        )
+
+    def run(self, params={}):
+        return self.connection.client.get_on_calls(
+            params.get(Input.SCHEDULEIDENTIFIER),
+            params.get(Input.SCHEDULEIDENTIFIERTYPE),
+            params.get(Input.FLAT),
+            params.get(Input.DATE),
+        )
```

Comment:
Would you be able to write this in the same format as the rest of the actions for consistency:
```data = {params.get(Input.x), etc etc
return client.function(data)```
Or is it done differently here for a reason?

Reply:
The reason is api get_on_calls has 4 specific parameters which we need to send in api. We passed data dictionary for api function if the request needs json body. Other actions don't need these parameters, that's why is differently.